packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayerContainer.js

```diff
@@ -16,24 +28,92 @@ function storeMapper (stores) {
 class InteractionLayerContainer extends Component {
   constructor () {
     super()
+
+    this.state = {
+      currentMark: null,
+      drawing: false,
+      markings: []
+    }
+
+    this.drawMark = this.drawMark.bind(this)
+    this.getEventOffset = this.getEventOffset.bind(this)
+    this.getNewMark = this.getNewMark.bind(this)
     this.onMouseDown = this.onMouseDown.bind(this)
     this.onMouseMove = this.onMouseMove.bind(this)
     this.onMouseUp = this.onMouseUp.bind(this)
   }
 
   componentDidMount () {
-    // TODO: We're simply logging the event stream here for now, but this will
-    // be passed to the active drawing tool for parsing
-    // const stream = this.props.drawing.eventStream
-    // stream.subscribe(z => console.log(z))
+    // const stream = this.props.eventStream
+    // stream.subscribe(z => console.log('z = ', z))
+    this.drawMark()
+  }
+
+  componentDidUpdate () {
+    this.drawMark()
+  }
+
+  drawMark () {
+    const { eventStream } = this.props
+
+    const stream = eventStream
+    stream.subscribe(event => {
+      const { currentMark, drawing, markings } = this.state
+      let newMark
+      // initial mark
+      if (markings.length === 0 && event.event === 'mousedown') {
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark, drawing: true })
+      }
+      // move mark while drawing
+      if (drawing && event.event === 'mousemove') {
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark })
+      }
+      if (drawing && event.event === 'mouseup') {
+        this.setState({ drawing: false })
+      }
+      // subsequent mark
+      if (!drawing && currentMark !== null && event.event === 'mousedown') {
+        const filteredMarkings = markings.filter(marking => marking.id !== currentMark.id)
+        filteredMarkings.push(currentMark)
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark, drawing: true, markings: filteredMarkings })
+      }
```

Comment:
I think each of the above could be their own class methods, `onMouseUp`, `onMouseDown` etc. If you want to avoid a switch statement, you could define an object that's set to call the event handlers:
```
const handlers = {
  mousedown: () => this.onMouseDown()
}
```

and call them as so:
```
handlers[event.type]()
```

Reply:
You want want to pass the original event object in as an argument in my example depending on if it is needed by the handler or any other data.